nexus_constructor/nexus/validation.py

```diff
@@ -0,0 +1,90 @@
+import attr
+import h5py
+from typing import Tuple, List, Optional
+
+
+class NexusFormatError(Exception):
+    pass
+
+
+@attr.s
+class ValidateDataset:
```

Comment:
Can we have some unit tests to cover this?

Reply:
It's covered, but not directly, agree it is worth direct tests as documentation.

Reply:
added tests in `test_nexus_validation.py`